otelcollector/prometheuscollector/tomlparser-ccp-default-scrape-settings.go

```diff
@@ -0,0 +1,146 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"regexp"
+)
+
+func (fcl *FilesystemConfigLoader) ParseConfigMapForDefaultScrapeSettings() (map[string]string, error) {
+	config := make(map[string]string)
+
+	if _, err := os.Stat(fcl.ConfigMapMountPath); os.IsNotExist(err) {
+		fmt.Println("configmap for ccp default scrape settings not mounted, using defaults")
+		return config, nil
+	}
+
+	content, err := ioutil.ReadFile(fcl.ConfigMapMountPath)
+	if err != nil {
+		return nil, fmt.Errorf("error reading config map file: %s", err)
+	}
+
+	lines := strings.Split(string(content), "\n")
+	for _, line := range lines {
+		parts := strings.SplitN(line, "=", 2)
+		if len(parts) == 2 {
+			config[strings.TrimSpace(parts[0])] = strings.TrimSpace(parts[1])
+		}
+	}
+
+	return config, nil
+}
+
+
+func (cp *ConfigProcessor) PopulateDefaultSettingValuesFromConfigMap(parsedConfig map[string]string) {
+	if val, ok := parsedConfig["controlplane-kube-controller-manager"]; ok && val != "" {
+		cp.ControlplaneKubeControllerManager = val
+		fmt.Printf("config::Using configmap scrape settings for controlplane-kube-controller-manager: %v\n", cp.ControlplaneKubeControllerManager)
+	}
+
+	if val, ok := parsedConfig["controlplane-kube-scheduler"]; ok && val != "" {
+		cp.ControlplaneKubeScheduler = val
+		fmt.Printf("config::Using configmap scrape settings for controlplane-kube-scheduler: %v\n", cp.ControlplaneKubeScheduler)
+	}
+
+	if val, ok := parsedConfig["controlplane-apiserver"]; ok && val != "" {
+		cp.ControlplaneApiserver = val
+		fmt.Printf("config::Using configmap scrape settings for controlplane-apiserver: %v\n", cp.ControlplaneApiserver)
+	}
+
+	if val, ok := parsedConfig["controlplane-cluster-autoscaler"]; ok && val != "" {
+		cp.ControlplaneClusterAutoscaler = val
+		fmt.Printf("config::Using configmap scrape settings for controlplane-cluster-autoscaler: %v\n", cp.ControlplaneClusterAutoscaler)
+	}
+
+	if val, ok := parsedConfig["controlplane-etcd"]; ok && val != "" {
+		cp.ControlplaneEtcd = val
+		fmt.Printf("config::Using configmap scrape settings for controlplane-etcd: %v\n", cp.ControlplaneEtcd)
+	}
```

Comment:
This isn't blocking and matters more when adding the regular scrape configs, but I was thinking we could have a struct to represent the settings of a default scrape job and then a list of those structs to group all of the default settings together to make it easier for partners to add more scrape configs. Something like below:

```
type DefaultScrapeJob struct {
  Name           string
  Enabled        bool
  ScrapeInterval string
  RegexKeepList  string
}

var DefaultScrapeJobs = []DefaultScrapeJob{
  DefaultScrapeJob{
    Name:           "kubelet",
    Enabled:        true,
    ScrapeInterval: DefaultScrapeJobInterval,
    RegexKeepList:  "kubelet_volume_stats_capacity_bytes|kubelet_volume_stats_used_bytes|kubelet_node_name|kubelet_running_pods|kubelet_running_pod_count|kubelet_running_sum_containers|kubelet_running_containers|kubelet_running_container_count|volume_manager_total_volumes|kubelet_node_config_error|kubelet_runtime_operations_total|kubelet_runtime_operations_errors_total|kubelet_runtime_operations_duration_seconds_bucket|kubelet_runtime_operations_duration_seconds_sum|kubelet_runtime_operations_duration_seconds_count|kubelet_pod_start_duration_seconds_bucket|kubelet_pod_start_duration_seconds_sum|kubelet_pod_start_duration_seconds_count|kubelet_pod_worker_duration_seconds_bucket|kubelet_pod_worker_duration_seconds_sum|kubelet_pod_worker_duration_seconds_count|storage_operation_duration_seconds_bucket|storage_operation_duration_seconds_sum|storage_operation_duration_seconds_count|storage_operation_errors_total|kubelet_cgroup_manager_duration_seconds_bucket|kubelet_cgroup_manager_duration_seconds_sum|kubelet_cgroup_manager_duration_seconds_count|kubelet_pleg_relist_interval_seconds_bucket|kubelet_pleg_relist_interval_seconds_count|kubelet_pleg_relist_interval_seconds_sum|kubelet_pleg_relist_duration_seconds_bucket|kubelet_pleg_relist_duration_seconds_count|kubelet_pleg_relist_duration_seconds_sum|rest_client_requests_total|rest_client_request_duration_seconds_bucket|rest_client_request_duration_seconds_sum|rest_client_request_duration_seconds_count|process_resident_memory_bytes|process_cpu_seconds_total|go_goroutines|kubernetes_build_info|kubelet_certificate_manager_client_ttl_seconds|kubelet_certificate_manager_client_expiration_renew_errors|kubelet_server_expiration_renew_errors|kubelet_certificate_manager_server_ttl_seconds|kubelet_volume_stats_available_bytes|kubelet_volume_stats_capacity_bytes|kubelet_volume_stats_inodes_free|kubelet_volume_stats_inodes_used|kubelet_volume_stats_inodes|kube_persistentvolumeclaim_access_mode|kube_persistentvolumeclaim_labels|kube_persistentvolume_status_phase",
  },
  DefaultScrapeJob{
    Name:           "coredns",
    Enabled:        true,
    ScrapeInterval: DefaultScrapeJobInterval,
    RegexKeepList:  "coredns_build_info|coredns_panics_total|coredns_dns_responses_total|coredns_forward_responses_total|coredns_dns_request_duration_seconds|coredns_dns_request_duration_seconds_bucket|coredns_dns_request_duration_seconds_sum|coredns_dns_request_duration_seconds_count|coredns_forward_request_duration_seconds|coredns_forward_request_duration_seconds_bucket|coredns_forward_request_duration_seconds_sum|coredns_forward_request_duration_seconds_count|coredns_dns_requests_total|coredns_forward_requests_total|coredns_cache_hits_total|coredns_cache_misses_total|coredns_cache_entries|coredns_plugin_enabled|coredns_dns_request_size_bytes|coredns_dns_request_size_bytes_bucket|coredns_dns_request_size_bytes_sum|coredns_dns_request_size_bytes_count|coredns_dns_response_size_bytes|coredns_dns_response_size_bytes_bucket|coredns_dns_response_size_bytes_sum|coredns_dns_response_size_bytes_count|coredns_dns_response_size_bytes_bucket|coredns_dns_response_size_bytes_sum|coredns_dns_response_size_bytes_count|process_resident_memory_bytes|process_cpu_seconds_total|go_goroutines|kubernetes_build_info",
  },
  DefaultScrapeJob{
    Name:           "cadvisor",
    Enabled:        true,
    ScrapeInterval: DefaultScrapeJobInterval,
    RegexKeepList:  "container_spec_cpu_quota|container_spec_cpu_period|container_memory_rss|container_network_receive_bytes_total|container_network_transmit_bytes_total|container_network_receive_packets_total|container_network_transmit_packets_total|container_network_receive_packets_dropped_total|container_network_transmit_packets_dropped_total|container_fs_reads_total|container_fs_writes_total|container_fs_reads_bytes_total|container_fs_writes_bytes_total|container_cpu_usage_seconds_total|container_memory_working_set_bytes|container_memory_cache|container_memory_swap|container_cpu_cfs_throttled_periods_total|container_cpu_cfs_periods_total|container_memory_rss|kubernetes_build_info",
  },
  DefaultScrapeJob{
    Name:           "kubeproxy",
    Enabled:        false,
    ScrapeInterval: DefaultScrapeJobInterval,
    RegexKeepList:  "kubeproxy_sync_proxy_rules_duration_seconds|kubeproxy_sync_proxy_rules_duration_seconds_bucket|kubeproxy_sync_proxy_rules_duration_seconds_sum|kubeproxy_sync_proxy_rules_duration_seconds_count|kubeproxy_network_programming_duration_seconds|kubeproxy_network_programming_duration_seconds_bucket|kubeproxy_network_programming_duration_seconds_sum|kubeproxy_network_programming_duration_seconds_count|rest_client_requests_total|rest_client_request_duration_seconds|rest_client_request_duration_seconds_bucket|rest_client_request_duration_seconds_sum|rest_client_request_duration_seconds_count|process_resident_memory_bytes|process_cpu_seconds_total|go_goroutines|kubernetes_build_info",
  },
  DefaultScrapeJob{
    Name:           "apiserver",
    Enabled:        false,
    ScrapeInterval: DefaultScrapeJobInterval,
    RegexKeepList:  "apiserver_request_duration_seconds|apiserver_request_duration_seconds_bucket|apiserver_request_duration_seconds_sum|apiserver_request_duration_seconds_count|apiserver_request_total|workqueue_adds_total|workqueue_depth|workqueue_queue_duration_seconds|workqueue_queue_duration_seconds_bucket|workqueue_queue_duration_seconds_sum|workqueue_queue_duration_seconds_count|process_resident_memory_bytes|process_cpu_seconds_total|go_goroutines|kubernetes_build_info|apiserver_request_slo_duration_seconds_bucket|apiserver_request_slo_duration_seconds_sum|apiserver_request_slo_duration_seconds_count",
  },
  DefaultScrapeJob{
    Name:           "kubestate",
    Enabled:        true,
    ScrapeInterval: DefaultScrapeJobInterval,
    RegexKeepList:  "kube_job_status_succeeded|kube_job_spec_completions|kube_daemonset_status_desired_number_scheduled|kube_daemonset_status_current_number_scheduled|kube_daemonset_status_number_misscheduled|kube_daemonset_status_number_ready|kube_deployment_status_replicas_ready|kube_pod_container_status_last_terminated_reason|kube_pod_container_status_waiting_reason|kube_pod_container_status_restarts_total|kube_node_status_allocatable|kube_pod_owner|kube_pod_container_resource_requests|kube_pod_status_phase|kube_pod_container_resource_limits|kube_replicaset_owner|kube_resourcequota|kube_namespace_status_phase|kube_node_status_capacity|kube_node_info|kube_pod_info|kube_deployment_spec_replicas|kube_deployment_status_replicas_available|kube_deployment_status_replicas_updated|kube_statefulset_status_replicas_ready|kube_statefulset_status_replicas|kube_statefulset_status_replicas_updated|kube_job_status_start_time|kube_job_status_active|kube_job_failed|kube_horizontalpodautoscaler_status_desired_replicas|kube_horizontalpodautoscaler_status_current_replicas|kube_horizontalpodautoscaler_spec_min_replicas|kube_horizontalpodautoscaler_spec_max_replicas|kubernetes_build_info|kube_node_status_condition|kube_node_spec_taint|kube_pod_container_info|kube_.*_labels|kube_.*_annotations|kube_service_info|kube_pod_container_status_running|kube_pod_container_status_waiting|kube_pod_container_status_terminated|kube_pod_container_state_started|kube_pod_created|kube_pod_start_time|kube_pod_init_container_info|kube_pod_init_container_status_terminated|kube_pod_init_container_status_terminated_reason|kube_pod_init_container_status_ready|kube_pod_init_container_resource_limits|kube_pod_init_container_status_running|kube_pod_init_container_status_waiting|kube_pod_init_container_status_restarts_total",
...
}
```

And then to set the settings for all default jobs from the configmap could be:

```
for _, job := range DefaultScrapeJobs {
  if enabledValue, ok := parsedConfig[job.Name]; ok && enabledValue != "" {
      job.Enabled = enabledValue
     fmt.Printf("config::Using configmap scrape settings for %s: %t\n", job.Name, enabledValue)
  }
}
```
And something similar could be done for the scrape interval and the keeplist